components/service/fretboard/src/test/java/mozilla/components/service/fretboard/FretboardTest.kt

```diff
@@ -146,4 +150,268 @@ class FretboardTest {
+        fretboard.setOverride(context, ExperimentDescriptor("first-name"), false)
+        fretboard.clearOverride(context, ExperimentDescriptor("first-name"))
+        verify(prefsEditor).remove("first-name")
+    }
+
+    @Test
+    fun testClearAllOverrides() {
+        val experimentSource = mock(ExperimentSource::class.java)
+        val experimentStorage = mock(ExperimentStorage::class.java)
+        val experiments = listOf(
+            Experiment("first-id",
+                name = "first-name",
+                match = Experiment.Matcher(
+                    appId = "test.appId"
+                )
+            )
+        )
+        `when`(experimentStorage.retrieve()).thenReturn(ExperimentsSnapshot(experiments, null))
+        val fretboard = Fretboard(experimentSource, experimentStorage)
+        fretboard.loadExperiments()
+
+        val context = mock(Context::class.java)
+        `when`(context.packageName).thenReturn("test.appId")
+        val sharedPrefs = mock(SharedPreferences::class.java)
+        val prefsEditor = mock(SharedPreferences.Editor::class.java)
+        `when`(prefsEditor.clear()).thenReturn(prefsEditor)
+        `when`(prefsEditor.putBoolean(ArgumentMatchers.anyString(), ArgumentMatchers.anyBoolean())).thenReturn(prefsEditor)
+        `when`(prefsEditor.putString(ArgumentMatchers.anyString(), ArgumentMatchers.anyString())).thenReturn(prefsEditor)
+        `when`(sharedPrefs.edit()).thenReturn(prefsEditor)
+        `when`(sharedPrefs.getBoolean(ArgumentMatchers.anyString(), ArgumentMatchers.anyBoolean())).thenAnswer { invocation -> invocation.arguments[1] as Boolean }
+        `when`(context.getSharedPreferences(ArgumentMatchers.anyString(), ArgumentMatchers.anyInt())).thenReturn(sharedPrefs)
+
+        val packageInfo = mock(PackageInfo::class.java)
+        packageInfo.versionName = "version.name"
+        val packageManager = mock(PackageManager::class.java)
+        `when`(packageManager.getPackageInfo(ArgumentMatchers.anyString(), ArgumentMatchers.anyInt())).thenReturn(packageInfo)
+        `when`(context.packageManager).thenReturn(packageManager)
+
+        assertTrue(fretboard.isInExperiment(context, ExperimentDescriptor("first-name")))
+        fretboard.setOverride(context, ExperimentDescriptor("first-name"), false)
+        fretboard.clearAllOverrides(context)
+        verify(prefsEditor).clear()
+    }
+
+    @Test
+    fun testUpdateExperimentsException() {
+        val source = mock(ExperimentSource::class.java)
+        doAnswer {
+            throw ExperimentDownloadException("test")
+        }.`when`(source).getExperiments(any())
+        val storage = mock(ExperimentStorage::class.java)
+        `when`(storage.retrieve()).thenReturn(ExperimentsSnapshot(listOf(), null))
+        val fretboard = Fretboard(source, storage)
+        fretboard.updateExperiments()
+    }
+
+    private fun <T> any(): T {
+        Mockito.any<T>()
+        return null as T
+    }
```

Comment:
We do have a method for this in the `support-test`. You can just add this to the test dependencies:

```
testImplementation ':support-test'
```

Reply:
Now it's using that method :)